### ISSUE TITLE
Read the single-line block of an at-rule in `declaration-block-single-line-max-declarations`

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#640]
### What is the problem you're trying to solve?

`declaration-block-single-line-max-declarations` (#639) walks rules alone, as Stylelint's own did: `root.walkRules`, the block of each rule printed and counted. So a single-line block of an at-rule is never read, however many declarations it holds:

```css
@font-face { font-family: x; src: y; }
@page { margin: 0; size: A4; }
@property --x { syntax: "<length>"; inherits: false; initial-value: 0; }
```

Under `1` every one of these passes, while `a { font-family: x; src: y; }` is reported. The rule's own suite pins the reading with an accept case naming `@font-face`, since the branch that brought the rule over carried it as it stood rather than widening it.

### What solution would you like to see?

Read every statement's block, rule or at-rule, since a block of declarations is what the rule is about and which keyword opens it is nothing the count depends on. `walkRules` plus `walkAtRules`, the at-rule passed over where it has no block (`hasBlock`), which covers a Less variable, a mixin call and every bodiless at-rule at once; `blockString` and `beforeBlockString` take either container already, so the index and the span come out as for a rule.

The change is a false negative fixed, so it files under `Changed` and the config goes major with it, which it does on almost every release anyway. Two things to settle in the branch: whether an at-rule holding rules rather than declarations on one line, `@media screen { a { color: pink; } }`, is a block of zero declarations to the rule, which the count answers on its own, and what each namespace's parser files inside the blocks of `@include`, `@mixin` and a Less detached ruleset, which a probe over the syntaxes answers before a line is written.
